util/loadutil/disk.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2023 The CubeFS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package loadutil
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/shirou/gopsutil/disk"
+)
+
+func getMatchCount(lhs string, rhs string) int {
+	count := len(lhs)
+	if count > len(rhs) {
+		count = len(rhs)
+	}
+	for i := 0; i < count; i++ {
+		if lhs[i] != rhs[i] {
+			return i + 1
+		}
+	}
+	return count
+}
+
+func GetMatchParation(path string) (*disk.PartitionStat, error) {
+	partitons, err := disk.Partitions(true)
+	if err != nil {
+		return nil, err
+	}
+	maxMatch := 0
+	matchParation := disk.PartitionStat{}
+	for _, partition := range partitons {
+		match := getMatchCount(path, partition.Mountpoint)
+		if match == len(partition.Mountpoint) && match > maxMatch {
+			matchParation = partition
+		}
+	}
+	return &matchParation, nil
+}
+
+var (
+	ErrInvaliPartition      = errors.New("invalid partiton")
+	ErrFailedToGetIoCounter = errors.New("failed to get io counter")
+)
+
+func getDeviceNameFromPartition(partition *disk.PartitionStat) (string, error) {
+	var name string
+	if n, _ := fmt.Sscanf(partition.Device, "/dev/%s", &name); n != 1 {
+		return "", ErrInvaliPartition
+	}
+	return name, nil
+}
+
+func GetIoCounter(partition *disk.PartitionStat) (*disk.IOCountersStat, error) {
```

Comment:
remove IOCounter, we have MetricIOBytes to record read/write bytes

Reply:
this function is necessary for getting ioutil.